ksql-functional-tests/src/main/java/io/confluent/ksql/test/tools/FakeKafkaRecord.java

```diff
@@ -54,20 +53,22 @@ public static FakeKafkaRecord of(
             ? ((ValueSpec)producerRecord.value()).getSpec()
             : producerRecord.value(),
         producerRecord.timestamp(),
-        getWindowData(producerRecord, windowType)
+        getWindowData(producerRecord)
     );
     return new FakeKafkaRecord(testRecord, producerRecord);
   }
 
+  @SuppressWarnings("unchecked")
   private static WindowData getWindowData(
-      final ProducerRecord producerRecord,
-      final WindowType windowType) {
+      final ProducerRecord producerRecord) {
     if (producerRecord.key() instanceof Windowed) {
       final Windowed windowed = (Windowed) producerRecord.key();
       return new WindowData(
```

Comment:
why not just have a constructor for `new WindowData(windowed)`

Reply:
Good option! Made the change!